ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md

```diff
@@ -2318,6 +2318,8 @@ sflow:
 
 ```yaml
 snmp_server:
+  engineid:
+    local: <local_engine_id>
```

Comment:
I think it would make more sense to use the plural for the keyname: "engine_ids" (because we can have one local and multiple remote ids) and add the parameters as a list:

```
snmp_server:
<snip>
  engine_ids:
    - type: < local | remote >
      name: < engine_name in hex >
      ip: < hostname | ip of remote engine >
      port: < udp-port of remote engine >
</snip>
```

Drawback: This allows to create multiple local ID's even only one is possible

Advantage: You can easily include the remote ID's

==

```
engine_ids:
  local:
  remote:
      -name: < engine_name in hex >
       ip: < hostname | ip of remote engine >
       port: < udp-port of remote engine >
```

Reply:
makes sense to me! 

for the remote's the logic would be quite easy, as you don't have to "cross" check if localized is true or not, I think you can easily add it it with this PR. Otherwise I can add it later once your PR is merged

Reply:
I have updated my PR with the discussed structure - I have not added the remote support for SNMPv3 users in it as there are some points I am not sure about - typically I can configure this and it will show up in the confg:

```
snmp-server user blah blahg remote 1.1.1.1 udp-port 1337 v3 localized 121212121212 auth sha 1234567890123456789012345678901234567890
```

but I am not sure how this is supposed to relate to a configured remote engine ID on the same device

Reply:
cool! skip the remote user for now (keep the remote engine config)

Reply:
@gmuloc the documentation is a bit confusing about the remote users. Let me try to sum it up:

first you will need to configure the remote engines name on the local system:
```
snmp-server engineID remote <REMOTE-IP> <REMOTENEGINEID>
```

We already have this in the DM (the list under remote: key)

==

Next you'll need to add the user of the remote agent to the local system:
```
snmp-server user <USERNAME> <GROUPNAME> remote <REMOTEIP> v3 auth sha <CLEARTEXT AUTHPW> priv aes <CLEARTEXT PRIV PW>
```

if you do it like this, EOS will use the REMOTENEGINEID from the previous command to encrypt the cleartexts.

you can also add the remote users by explicitly outting the REMOTEENGINEID in localization:
```
snmp-server user <USERNAME> <GROUPNAME> remote <REMOTEIP> v3 localized <REMOTENEGINEID> auth sha <CLEARTEXT AUTHPW> priv aes <CLEARTEXT PRIV PW>
```

==

the remote users are needed to send snmp informs (instead of traps) using snmpv3.

@ClausHolbechArista et al. Please correct me if I'm wrong about this.

Reply:
Thanks @ccsnw  - so essentially this is not too difficult to add for the `remote <REMOTEIP> udp-port <UDP-PORT>` part in the user declaration - the only thing is that it then requires also to change the `localized` variable as it is now possible to give any value to it - I initially used a boolean to be able to pick up the local engine ID and automatically regenerate the correct user CLI (well the keys still need to be changed) when changing the engine ID:
```
    users:
      - name: < username >
        group: < group_name >
        version: < v1 | v2c | v3 >
        localized: < true | false >
```

So now I am thinking that it should look like:
```
    users:
      - name: < username >
        group: < group_name >
        # remote_ip and udp_port are used for remote users
        remote_ip: < ip_address >
        # udp_port will not be used if no remote_ip is configured
        udp_port: < udp_port >
        version: < v1 | v2c | v3 >
        localized: < engine_name in hex >
```
and just like for the regular CLI the user is in charge of giving the correct value there.

Reply:
@gmuloc yup, will work if you turn localized into a hex string (also did it like that in a PR that I had prepared before seeing yours)

if the user doesn't provide the hex for the localized key, they pw's are expected in cleartext. Think you can add this to the readme if its not in there yet 

"and just like for the regular CLI the user is in charge of giving the correct value there."
>> Yes, but the Arista guys are working on input validation for v4 of AVD

Reply:
Ack - changes pushed in the latest commit

> if the user doesn't provide the hex for the localized key, they pw's are expected in cleartext. Think you can add this to the readme if its not in there yet

Yeah that's in the README

> Yes, but the Arista guys are working on input validation for v4 of AVD

hehe - I guess I should take a look at what's going on there as well